adala/skills/_base.py

```diff
@@ -548,30 +589,22 @@ class AnalysisSkill(Skill):
     Analysis skill that analyzes a dataframe and returns a record (e.g. for data analysis purposes).
     See base class Skill for more information about the attributes.
     """
-
+    input_prefix: str = ""
     input_separator: str = "\n"
     chunk_size: Optional[int] = None
 
-    def apply(
-        self,
-        input: Union[InternalDataFrame, InternalSeries, Dict],
-        runtime: Runtime,
-    ) -> InternalDataFrame:
-        """
-        Applies the skill to a dataframe and returns a record.
-
-        Args:
-            input (InternalDataFrame): The input data to be processed.
-            runtime (Runtime): The runtime instance to be used for processing.
+    def _iter_over_chunks(self, input: InternalDataFrame, chunk_size: Optional[int] = None):
 
-        Returns:
-            InternalSeries: The record containing the analysis results.
-        """
+        if input.empty:
```

Comment:
this one is a wip, the chunk iteration flow will be improved within the current PR

Reply:
https://humansignal.atlassian.net/browse/DIA-1489